astrix-gs/src/main/java/com/avanza/astrix/gs/SpaceTaskDispatcher.java

```diff
@@ -127,7 +127,7 @@ public int partitionCount() {
 		usingErrorReporter(subscriber, serviceUnavailable()).accept(() -> {
 			// Submit task on current thread in executorService
 			AsyncFuture<T> taskResult = gigaSpace.execute(task, routingKey);
-			GsUtil.subscribe(taskResult, subscriber);
+			GsUtil.subscribe(taskResult, subscriber, contextPropagation);
```

Comment:
I'm wondering if we should consider putting this behind a feature toggle? What do you think @oskarwiksten ?

Reply:
Yes, let's do that. Have added a separate commit for this now.

Reply:
nice! All good for merge